src/generate.ts

```diff
@@ -304,38 +345,129 @@ class Generator {
   }
 }
 
+export interface GenerateOptions {
+  types: boolean
+  request: boolean
+  operations: boolean
+  prettier: boolean
+  withDoc: boolean
+  patchScript?: string
+  onParsed?: (parsed: ParsedOpenApi) => void
+}
+
+const componentTypes = [
+  "schemas",
+  "responses",
+  "examples",
+  "requestBodies",
+  "headers",
+  "securitySchemes",
+  "links",
+  "callbacks"
+]
+
 export async function generate(
-  docOrPathToDoc: string | OpenAPIV3.Document
+  docPath: string | string[],
+  generateOptions: Partial<GenerateOptions> = {}
 ): Promise<string> {
-  const doc = (await bundle(docOrPathToDoc)) as OpenAPIV3.Document
-
-  const generator = new Generator(doc)
+  let doc: OpenAPIV3.Document
+  if (!Array.isArray(docPath)) {
+    docPath = [docPath]
+  }
+  for (const path of docPath) {
+    const newDoc = (await SwaggerParser.bundle(path)) as OpenAPIV3.Document
+    if (!doc) {
+      doc = newDoc
+      if (!doc.components) {
+        doc.components = {}
+      }
+    } else {
+      // merge paths, do not override
+      for (const [path, pathItemObj] of Object.entries(newDoc.paths)) {
+        if (!doc.paths[path]) {
+          doc.paths[path] = pathItemObj
+        }
+      }
+      // merge types
+      if (newDoc.components) {
+        for (const componentType of componentTypes) {
+          if (newDoc.components[componentType]) {
+            for (const [key, val] of Object.entries(
+              newDoc.components[componentType]
+            )) {
+              if (!doc.components[componentType]) {
+                doc.components[componentType] = {}
+              }
+              if (!doc.components[componentType][key]) {
+                doc.components[componentType][key] = val
+              }
+            }
+          }
+        }
+      }
+    }
+    if (generateOptions.patchScript) {
+      const patchScript = require(generateOptions.patchScript)
+      doc = await patchScript.patch(doc, SwaggerParser)
```

Comment:
this seems overly risky - just pulling in a random script and executing. What is a use case for this?

Reply:
swagger document is fixed in https://github.com/influxdata/influxdb-client-js/blob/master/packages/apis/scripts/patchSwagger.js so that it contains also resource values from the cloud spec